### ISSUE TITLE
face_seg算法迁移到module

### ISSUE THREAD

**Issue by JiaojiaoYe1994** [Oneflow-Inc/models#13]
### 安排计划

* 6月４日，任务分工，缺失算子统计
  - ✅ 算子统计参加下表
  - ✅ 任务分工： 算法部分迁移：魏日令， 算子补齐：叶娇娇
* 6月5日- 6月11日前，完成推断脚本的迁移（包括数据，推理，模型）和预训练模型的转换@魏日令 @weirl-922，和推理所需算子的迁移 ＠叶娇娇 ＠杜承垚 ＠石永涛
  - ✅ 推理脚本迁移至新接口
  - ✅ 预训练模型转换至oneflow模式格式
  - ✅ 推理（inference & model）所需算子迁移至module  
* 6月12 -  6月18日，计划完成推理脚本的测试，训练所需算子的开发
  - ✅ 训练算子都merged
  - ✅ 推理调试成功
  - 
* 6月19 - 6月25日, 　，提交pr 1. @JiaojiaoYe1994 完善代码，添加comment, readme, 2. @weirl-922  完成train部分，对齐精度

### 链接和参考

* model 合并分支: https://github.com/Oneflow-Inc/models/

* 算子统计和进度汇报统计: https://github.com/Oneflow-Inc/models/issues/3

* pytorch 版本算法原实现：https://github.com/nasir6/face-segmentation

* pytorch 版本完善: https://github.com/JiaojiaoYe1994/face-segmentation

* oneflow lazy 版本实现：https://github.com/Oneflow-Inc/oneflow_vision_model/tree/main/face_seg

  

### 算子统计

FaceSeg 算法迁移所需算子
|      | 所需算子                                  | 对齐目标                                                     | 目标算子             | 备注                                | 认领人 |
| ---- | ----------------------------------------- | ------------------------------------------------------------ | -------------------- | ----------------------------------- | ------ |
model.py | &nbsp; | &nbsp; | &nbsp; | &nbsp; | &nbsp; | &nbsp;
&nbsp; | nn.ConvTranspose2d | flow.nn.torch_conv2d_transpose | 缺失 | merged https://github.com/Oneflow-Inc/oneflow/pull/5101  | 张晓雨;
&nbsp; | torchvision.models.resnet34 | &nbsp; | &nbsp; | &nbsp; | &nbsp;
Inference.py | &nbsp; | &nbsp; | &nbsp; | &nbsp; | &nbsp;
&nbsp; | transforms.Compose | &nbsp; | 缺失 | 用numpy处理代替 | &nbsp;
&nbsp; | transforms.Resize | &nbsp; | 缺失 | 用numpy处理代替 | &nbsp;
&nbsp; | transforms.Normalize | &nbsp; | 缺失 | 用numpy处理代替 | &nbsp;
&nbsp; | transforms.Resize | &nbsp; | 缺失 | 用numpy处理代替| &nbsp;
validation.py | &nbsp; | &nbsp; | &nbsp; | &nbsp; | &nbsp;
&nbsp; | torch.no_grad | &nbsp; |  已迁移 | &nbsp; | &nbsp;
train.ipynb | &nbsp; | &nbsp; | &nbsp; | &nbsp; | &nbsp;
&nbsp; | nn.BCEWithLogitsLoss | &nbsp; | 缺失 | https://github.com/Oneflow-Inc/oneflow/pull/5173/ | 叶娇娇 @JiaojiaoYe1994 ;
&nbsp; | torch.utils.data.DataLoader | &nbsp; | &nbsp; | &nbsp; | &nbsp;
&nbsp; | torch.float | &nbsp; | 已迁移 | &nbsp; | &nbsp;
&nbsp; | optim.lr_scheduler.CosineAnnealingLR | &nbsp; | 有CosineSheduler，功能一样 | https://github.com/Oneflow-Inc/oneflow/pull/5112 |三土 @duchengyao 

<br class="Apple-interchange-newline">

**Comment by duchengyao** [comment]
* BCEWithLogitsLoss 用 nill loss 就可以了
* ConvTranspose2d 已经有人在搬运
